astropy/coordinates/builtin_frames/icrs_cirs_transforms.py

```diff
@@ -129,12 +129,10 @@ def cirs_to_icrs(cirs_coo, icrs_frame):
                                               distance=cirs_coo.distance,
                                               copy=False)
 
-        newxyz = intermedrep.to_cartesian().xyz
-        # roll xyz to last axis and add the barycentre position
-        newxyz = np.rollaxis(newxyz, 0, newxyz.ndim) + astrom['eb'] * u.au
-        # roll xyz back to the first axis
-        newxyz = np.rollaxis(newxyz, -1, 0)
-        newrep = CartesianRepresentation(newxyz).represent_as(SphericalRepresentation)
+        astrom_eb = CartesianRepresentation(
+            u.Quantity(np.rollaxis(astrom['eb'], -1, 0), u.au, copy=False),
+            copy=False)
```

Comment:
Yes, I wondered also whether to make the `CartesianRepresentation` initializer a little more flexible, so one could give an array and just tell it which axis to take as the xyz one (rather than only allowing axis 0). One might then also want to be able to pass in a `unit`...